### ISSUE TITLE
envdte package shows up as soon as Tmds.ExecFunction is installed

### ISSUE THREAD

**Issue by tisis2** [tmds/Tmds.ExecFunction#16]
since 0.6.0 in the package explorer of VS the envdte package will appear, since this is not added as a dependency but by targets file as its own package. this will also cause this package to show for updates. 

<img width="933" alt="image" src="https://user-images.githubusercontent.com/18416393/187218611-efe48a30-4488-4465-b8a9-9ec2acbaa93a.png">

if there is really no other possibility to auto attach debugger and solve the dependencies, all this should be at least in its own package.
maybe there should be an auto attach package that uses that envdte as dependency, so you dont need any copy tasks and this package can only be installed when debug config is chosen

**Comment by Bertk** [comment]
Hi,
we have also some trouble with the PackageReference for EnvDTE in [Tmds.ExecFunction.targets](https://github.com/tmds/Tmds.ExecFunction/blob/15b5456a49c2d2bd34b59dc8e7c88bc34ac674a0/src/Tmds.ExecFunction/Tmds.ExecFunction.targets#L3C6-L3C66). It should use `PrivateAssets="all"` to avoid  Nuget restore loops.

By the way, when will the EnvDTE version upgraded to [17.7.37355](https://www.nuget.org/packages/envdte/17.7.37355) (VS2022 version).

<img width="146" alt="image" src="https://github.com/tmds/Tmds.ExecFunction/assets/2315215/9d6b5642-2223-47be-97f1-6408c3be2ef1">

**Comment by tmds** [comment]
I won't be looking into this myself as not that many people have responded to this issue so far, and I don't want to break VS users. Because I don't use VS, I can't validate the change.

If someone has a fix for this, feel free to suggest it in a PR.

> By the way, when will the EnvDTE version upgraded to [17.7.37355](https://www.nuget.org/packages/envdte/17.7.37355) (VS2022 version).

I'll be upgrading this for an upcoming release.

**Comment by Bertk** [comment]
FYI. We can currently not use .NET 7.0 feature ([Central package management](https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-7#central-package-management) for a test project which uses Tmds.ExecFunction because this triggers a nuget restore loop in VS2022.

**Comment by tmds** [comment]
I'm open to PRs to fix and improve this, but as a non-VS user, I can not help provide them.

**Comment by tmds** [comment]
I had accidentally broken the vs code debugging in v0.7.0.
While fixing it, I've added some bits that should make it work with central package management.
This is part of v0.7.1 which was just published to nuget.org.

**Comment by tmds** [comment]
@Bertk when you had a change to try 0.7.1, please let me know if the central package management issue is fixed.

**Comment by Bertk** [comment]
@tmds Yes, I could activate central package management for the test and now also use VS2022. Thanks.

**Comment by smaillet** [comment]
I think the issue here (at least from me) is that it appears as a TOP-LEVEL package instead of as a dependency. It is VERY frustrating to have a mysterious dependency show in the UI as top level but NOTHING in the project actually references it. ALL other packages show dependencies of the package as children so it is easy to identify the source of the inclusion (For rapid CVE response etc...)

NOTE: After digging into it the problem is that `Tdms.ExeFunction.targets` includes a `PackageReference` element. Implicitly referenced props/targets files should NEVER do that! The source of the reference is extremely difficult to find when dealing with updates/CVEs. Instead, the package itself should have any dependency information. I don't know what such a reference is for as there's no docs, but it does seem to be conditioned on `EnableExecFunctionVsDebugging` property (again no idea what that is as there's no docs) but setting that to false at least removes the mystery dependency.
